src/GeneratorV2/Process/Overloaders/VectorOverloader.cs

```diff
@@ -0,0 +1,188 @@
+                            countParameters.Add(countParameter);
+                        }
+                    }
+
+                    if (length < 2 || length > 4)
+                    {
+                        continue;
+                    }
+
+                    // Save the new parameter.
+                    nameTable.Rename(parameter, parameter.Name + "_ptr");
+                    string vectorType = "Vector" + length + (typeChar == 'f' ? "" : typeChar);
+                    newParameters[i] = parameter with
+                    {
+                        Type = new CSRef(csType.Constant ? CSRef.Type.In : CSRef.Type.Ref, new CSType(vectorType, csType.Constant)),
+                        Length = null
+                    };
+                    foundVector = true;
+                    pointerParameters.Add(parameter);
+                    vectorParameters.Add(newParameters[i]);
+                }
+            }
+
+            if (foundVector)
+            {
+                // Remove the count parameters from the new overload.
+                for (var i = countParameters.Count - 1; i >= 0; i--)
+                {
+                    Parameter parameterToRemove = countParameters[i];
+                    newParameters.Remove(parameterToRemove);
+                }
+
+                // Get the new name for the overload.
+                int vectorStartIndexInName = vIndex - 1;
+                if (char.IsDigit(name[vIndex - 2]))
+                {
+                    vectorStartIndexInName -= 1;
+                }
+                string newName = name[..(vectorStartIndexInName)] + name[(vIndex+1)..];
+
+                newOverloads = new List<Overload>()
+                {
+                    overload with
+                    {
+                        InputParameters = newParameters.ToArray(),
+                        NameTable = nameTable,
+                        NestedOverload = overload,
+                        OverloadName = newName,
+                        MarshalLayerToNested = new VectorOverloadLayer(pointerParameters, vectorParameters, countParameters),
+                    },
+                    overload,
+                };
+                return true;
+            }
+
+            newOverloads = null;
+            return false;
+        }
+
+        public class VectorOverloadLayer : IOverloadLayer
+        {
+            private readonly List<Parameter> _pointerParameters;
+            private readonly List<Parameter> _vectorParameters;
+            private readonly List<Parameter> _countParameters;
+
+            public VectorOverloadLayer(List<Parameter> pointerParameters,
+                List<Parameter> vectorParameters, List<Parameter> countParameters)
+            {
+                _pointerParameters = pointerParameters;
+                _vectorParameters = vectorParameters;
+                _countParameters = countParameters;
+            }
+
+            private CsScope _scope;
```

Comment:
Nitpick: To match the style of the other overloaders this should be named `Scope`.